packages/core/src/interpreter.ts

```diff
@@ -402,7 +385,7 @@ export class Interpreter<
 
     const resolvedState =
       initialState === undefined
-        ? this.initialState
```

Comment:
any particular reason behind the change here?

Reply:
I think this was important because the `initialState` should be cached at the interpreter level

Reply:
Reverted: eab43a59bab69666479e402ee9f4f599e80b1e36